election_predictor/interface/main.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+import pandas as pd
+from datetime import datetime, timedelta
+
+# Import params
+from election_predictor.params import *
+
+# Import all functions inside data.py from ml_logic
+from election_predictor.ml_logic.data import fetch_clean_data
+
+# Import preprocessor
+from election_predictor.ml_logic.preprocessor import preprocessor
+
+# Import modelling functions from ml_logic
+from election_predictor.ml_logic.model import XGBoostModel
+
+def predict_election() -> dict:
```

Comment:
should this take election_year as an argument?

Reply:
For now the function is designed to predict the 2024 election only. Next step will be for it to take previous election years also as the UI will need to fetch results for these years (if we stick to current design). My aim is to have live data cleaned, fetched, and predicted on for 2024 as a sort of MVP and then go from there.

Additionally, logic for evaluating model performance against previous election results is yet to be added but should be to optimise for the most performant model as live data sourcing could alter performance.